android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -351,11 +356,28 @@ class GeckoWebExtension(
         return nativeExtension.metaData.let {
             Metadata(
                 name = it.name,
+                fullDescription = it.fullDescription,
+                downloadUrl = it.downloadUrl,
+                updateDate = it.updateDate,
+                averageRating = it.averageRating.toFloat(),
+                reviewCount = it.reviewCount,
                 description = it.description,
                 developerName = it.creatorName,
                 developerUrl = it.creatorUrl,
                 homePageUrl = it.homepageUrl,
+                creatorName = it.creatorName,
+                creatorUrl = it.creatorUrl.orEmpty(),
+                reviewUrl = it.reviewUrl,
                 version = it.version,
+                getIcon = {
```

Comment:
Should we add an argument to pass the image size? instead of having it hardcoded?

Reply:
If this is a (dynamic) property, maybe it should be named `icon` or `icon48` if we cannot pass an argument to specify the version.